### PR TITLE
Include csapp.c and csapp.h files from textbook website

### PR DESCRIPTION
**Pull request by frozenfrank**
## Overview
The class textbook, slides, and examples frequently uses custom methods from `cs_app.h` that automatically handle UNIX errors.

As students, we should be able to follow the best practices and use these files ourselves.

## Work List
This is a draft PR because multiple things must be done:
 - [x] Include the raw files
 - [ ] Include the files from project starter files
 - [ ] Rename/remove duplicated sections that have name conflicts with code in the files

## External Work List
After the class repo is internally consistent with these changes, external changes may be required as well:
- [ ] Ensure the AutoGrading tool works properly with these new files

### CODE REVIEW

**Comment by cdeccio** [comment]
There are various reasons for which we do not reference or use the textbook's csapp files.  These include the following:
 - Some of their utility functions are not as universal as they seem to claim.  For example, if a certain system call is issued, sometimes the right call is to exit with non-zero status, and sometimes it is not.  It depends on the program, the place in the program, and more.
 - Creating a utility function is good practice, but more important in this class is that the students know what is happening in the utility function.  I would actually rather they copy/paste it a dozen times and know what it means that call it a dozen times and not really know what it means.  Of course, ideally, they would write it themselves and call it, but that is still different.
 - When transitioning between Lab 4 and Lab 5 (and probably Lab 3 also), it is especially important that students have better control than what is available in the utility functions provided by the csapp files.
 
Short version: there is no plan to add these files.  But if there is a place where we could be more clear about the fact that we are not using these, we can add that.

**Comment by frozenfrank** [comment]
Closing this PR because I don't intend to finish this process myself. A future person can reference this if they so desire to continue forward.
